module/x/gravity/keeper/pool.go

```diff
@@ -16,8 +16,20 @@ import (
 // - checks a counterpart denominator exists for the given voucher type
 // - burns the voucher for transfer amount and fees
 // - persists an OutgoingTx
-// - adds the TX to the `available` TX pool via a second index
-func (k Keeper) AddToOutgoingPool(ctx sdk.Context, sender sdk.AccAddress, counterpartReceiver string, amount sdk.Coin, fee sdk.Coin) (uint64, error) {
+// - adds the TX to the `available` TX pool
+func (k Keeper) AddToOutgoingPool(
+	ctx sdk.Context,
+	sender sdk.AccAddress,
+	counterpartReceiver string,
+	amount sdk.Coin,
+	fee sdk.Coin,
+) (uint64, error) {
+	if ctx.IsZero() || sender.Empty() || counterpartReceiver == "" ||
```

Comment:
If there is a more elegant way to handle parameter checking in Go I would love to adjust this. Additionally, with this approach we do not tell the method caller which argument was a problem. I have left the parameter checking in this style because it causes less method bloat than individual checks like `if amount.Amount.IsNil() || ... { return 0, "Bad amount argument" }`

Reply:
the only better way I can think of is having an error checking function specifically for this, just moving the bloat elsewhere where it won't be distracting.

It does seem like the most 'Go' way to do this.

Reply:
For several of these (for example the coin type) there could be verifier functions that we get to reuse here.